src/couch_replicator_scheduler.erl

```diff
@@ -404,3 +410,97 @@ update_history(Job, Type, When, State) ->
     History0 = [{Type, When} | Job#job.history],
     History1 = lists:sublist(History0, State#state.max_history),
     Job#job{history = History1}.
+
+
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+
+
+backoff_msec_test() ->
+    BaseMicros = ?BACKOFF_INTERVAL_MICROS,
+    ?assertEqual(backoff_msec(1), BaseMicros),
+    ?assertEqual(backoff_msec(2), 2 * BaseMicros),
+    ?assertEqual(backoff_msec(3), 4 * BaseMicros),
+    MaxBackoff = (1 bsl ?MAX_BACKOFF_EXPONENT) * BaseMicros,
```

Comment:
since we know the bounds, we can use a test generator to assert all the values;

    [?_assertEqual(Penalty * ?BACKOFF_INTERVAL_MICROS, backoff_penalty_micros(crashes(Crashes)))
     || {Crashes, Penalty} <-
            [{1, 1}, {2, 2}, {3, 4}, {4, 8}, {5, 16}, {6, 32}, {7, 64},
             {8, 128}, {9, 256}, {10, 512}, {11, 1024}, {12, 1024}]].